packages/stream-tile/src/tile-document.ts

```diff
@@ -173,9 +169,7 @@ export class TileDocument<T = Record<string, any>> extends Stream {
       // document as there shouldn't be any existing state for this doc on the network.
       opts.syncTimeoutSeconds = 0
     }
-    const commit = genesisCommit.data
-      ? await _signDagJWS(ceramic, genesisCommit, genesisCommit.header.controllers[0])
-      : genesisCommit
+    const commit = genesisCommit.data ? await _signDagJWS(ceramic, genesisCommit) : genesisCommit
```

Comment:
What's the behavior now if the currently authenticated DID isn't the controller of this stream?  We should make sure that an exception is thrown rather than having the commit just silently fail to apply (or worse actually get applied, but I assume that somewhere else in commit application code will prevent that)

Reply:
Can you also check to confirm that there's a test for trying to author commits on streams you don't own?  I assume there must be one already but would be good to make sure

Reply:
Agreed that it would be good to have this test. My assumption is that we should throw an error when the commit is applied! @ukstv

Reply:
Ah, that's what you said Spencer :)

Reply:
> What's the behavior now if the currently authenticated DID isn't the controller of this stream?

It throws. Just tried that with my test harness. After signing commit, control flow goes through `applyCommit` function that verifies the signature.

Reply:
Can not add a test here, because there is no resolver that can be used to demonstrate the behaviour.

Reply:
> Can not add a test here, because there is no resolver that can be used to demonstrate the behaviour.

Sure you can. Just create a Tile with key-did-1 then try to update it with key-did-2?

Reply:
> Sure you can. Just create a Tile with key-did-1 then try to update it with key-did-2?

Hmm, this does not get deep enough for 'controllers' code path, but yeah, makes sense.

Reply:
Ooh, we should also test this for *creating* a stream with the wrong controller, not just for updates.  I just realized that I suspect my fix from https://github.com/ceramicnetwork/js-ceramic/pull/1597 may not have handled the stream **creation** case properly and may cause the exception to be swallowed.  If so the fix would be to do the same for stream creation as we do for stream updates and pass the `throwOnInvalidCommit` flag in `opts`

Reply:
Done.

Reply:
nice!  Glad to see I didn't accidentally break stream creation 😅.  And I see now why it works so it makes sense too.